src/core/search/sort_indices.cc

```diff
@@ -0,0 +1,61 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "core/search/sort_indices.h"
+
+#include <absl/container/flat_hash_set.h>
+#include <absl/strings/ascii.h>
+#include <absl/strings/numbers.h>
+#include <absl/strings/str_split.h>
+
+#include <algorithm>
+
+namespace dfly::search {
+
+using namespace std;
+
+namespace {}  // namespace
+
+template <typename T>
+void SimpleValueSortIndex<T>::Sort(std::vector<DocId>* entries, std::vector<ResultScore>* out,
+                                   size_t limit, bool desc) const {
+  auto cb = [this, desc](const auto& lhs, const auto& rhs) {
+    return desc ? (values_[lhs] > values_[rhs]) : (values_[lhs] < values_[rhs]);
+  };
+  std::partial_sort(entries->begin(), entries->begin() + std::min(entries->size(), limit),
+                    entries->end(), cb);
+
+  out->clear();
+  out->reserve(entries->size());
+  for (auto id : *entries)
+    out->push_back(values_[id]);
```

Comment:
`out` is used for `scores_`, is it ok, that `scores` will be partially sorted as well? Based on my understanding it's fine, I am just double checking

Reply:
Yes, actually on a second thought we don't need to serialize all the scores, only the partial sort prefix 🤓 Will fix it